internal/provider/resource_workspace_current_state.go

```diff
@@ -0,0 +1,313 @@
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	tharsis "gitlab.com/infor-cloud/martian-cloud/tharsis/tharsis-sdk-go/pkg"
+	sdktypes "gitlab.com/infor-cloud/martian-cloud/tharsis/tharsis-sdk-go/pkg/types"
+)
+
+const (
+	jobCompletionPollInterval = 5 * time.Second
+)
+
+var (
+	applyRunComment = "terraform-provider-tharsis" // must be var, not const, to take address
+)
+
+// WorkspaceCurrentStateModel is the model for a workspace_current_state.
+// Please note: Unlike many/most other resources, this model does not exist in the Tharsis API.
+// The workspace path, module source, and module version uniquely identify this workspace_current_state.
+type WorkspaceCurrentStateModel struct {
+	WorkspacePath types.String `tfsdk:"workspace_path"`
+	ModuleSource  types.String `tfsdk:"module_source"`
+	ModuleVersion types.String `tfsdk:"module_version"`
+	Teardown      types.Bool   `tfsdk:"teardown"`
+}
+
+// Set the Teardown field's default value of false.
+func (m *WorkspaceCurrentStateModel) setDefaultTeardown() {
+	if m.Teardown.IsNull() || m.Teardown.IsUnknown() {
+		m.Teardown = types.BoolValue(false)
+	}
+}
+
+// Ensure provider defined types fully satisfy framework interfaces
+var (
+	_ resource.Resource                = (*workspaceCurrentStateResource)(nil)
+	_ resource.ResourceWithConfigure   = (*workspaceCurrentStateResource)(nil)
+	_ resource.ResourceWithImportState = (*workspaceCurrentStateResource)(nil)
+)
+
+// NewWorkspaceCurrentStateResource is a helper function to simplify the provider implementation.
+func NewWorkspaceCurrentStateResource() resource.Resource {
+	return &workspaceCurrentStateResource{}
+}
+
+type workspaceCurrentStateResource struct {
+	client *tharsis.Client
+}
+
+// Metadata returns the full name of the resource, including prefix, underscore, instance name.
+func (t *workspaceCurrentStateResource) Metadata(ctx context.Context,
+	req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = "tharsis_workspace_current_state"
+}
+
+func (t *workspaceCurrentStateResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	description := "Defines and manages a workspace current state."
+
+	resp.Schema = schema.Schema{
+		Version:             1,
+		MarkdownDescription: description,
+		Description:         description,
+		Attributes: map[string]schema.Attribute{
+			"workspace_path": schema.StringAttribute{
+				MarkdownDescription: "The full path of the workspace.",
+				Description:         "The full path of the workspace.",
+				Required:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
```

Comment:
Some of these options shouldn't need the `UseStateForUnknown` since they will always be defined.